chapter06/src/main/java/com/seaofnodes/simple/node/ScopeNode.java

```diff
@@ -124,15 +124,15 @@ public ScopeNode dup() {
      * innermost stack level where the name is bound.
      *
      * @param that The ScopeNode to be merged into this
-     * @return A new Region node representing the merge point
+     * @return A new node representing the merge point
      */
     public Node mergeScopes(ScopeNode that) {
-        RegionNode r = (RegionNode) ctrl(new RegionNode(null,ctrl(), that.ctrl()).keep().peephole());
```

Comment:
@cliffclick I didn't do this because we don't assign a type to a Node until peephole() is invoked, so we end up with a null _type in Region at this point. Any thoughts on whether all Nodes should get a default _type on construction?

Reply:
yeah; Node constructor can set BOTTOM or "compute" (except for constructors with more to do before compute() like Constant)